Added Firebase App distribution doc (#407)
appcircleio/appcircle-docusaurus: docs/workflows/common-workflow-steps/firebase-app-distribution.md

---
title: Firebase App Distribution
description: Firebase App Distribution, enables developers to distribute their applications to testers and stakeholders for testing purposes.
tags: [firebase, mobile, distribution]
sidebar_position: 8
---

import Screenshot from '@site/src/components/Screenshot';

# Firebase App Distribution

> [**Firebase App Distribution**](https://firebase.google.com/docs/app-distribution) is a platform provided by [Google Firebase](https://firebase.google.com) that allows developers to distribute pre-release versions of their mobile apps to testers and stakeholders for testing purposes. It simplifies the process of distributing Android and iOS apps, providing features such as targeted distribution, release notes, feedback collection, and more.

Reviewer: ```suggestion
[**Firebase App Distribution**](https://firebase.google.com/docs/app-distribution), offered by [Google Firebase](https://firebase.google.com/), enables developers to distribute pre-release versions of mobile apps to testers and stakeholders. This platform streamlines the distribution of Android and iOS apps and offers features like targeted distribution, release notes, and feedback collection.
```

> The Appcircle **Firebase App Distribution** step enables you to efficiently distribute your mobile applications to testers and relevant parties directly from your Appcircle workflow. With this integration, you can optimize the distribution process and gather valuable feedback to iterate on your app before its public release.

Reviewer: ```suggestion
The Appcircle **Firebase App Distribution** step facilitates the efficient distribution of mobile applications to testers directly through the Appcircle workflow. This integration helps optimize the distribution process and collect valuable feedback for app iterations before public release.
```

:::caution

Please note that you can also distribute your app via Appcircle. Utilizing Appcircle's distribution modules enhances manageability within the platform.

For more details, please refer to the following links:

- [Appcircle Testing Distribution](https://docs.appcircle.io/distribute/)
- [Appcircle Enterprise App Store](https://docs.appcircle.io/enterprise-appstore/)
- [Appcircle Publish](https://docs.appcircle.io/publish-module/)

:::

### Prerequisites

> The workflow steps that need to be executed before running the **Firebase App Distribution** workflow step, along with their respective reasons, are listed in the table below. Prerequisites vary depending on the platform used:

Reviewer: ```suggestion
Below are the workflow steps required before running the **Firebase App Distribution** step, listed with their reasons. Prerequisites vary by platform:
```

#### For Android (Java / Kotlin and React Native) 

| Prerequisite Workflow Step        | Description                                                                                              |
| --------------------------------- | -------------------------------------------------------------------------------------------------------- |
| [**Android Build**](https://docs.appcircle.io/workflows/android-specific-workflow-steps/android-build) | Generates the app required for the **Firebase App Distribution** step.                                                                           |

> | [**Android Sign**](https://docs.appcircle.io/workflows/android-specific-workflow-steps/android-sign)   | Required if using a signed app. Processes the output for signing. If already signed, this step can be skipped. |

Reviewer: ```suggestion
| [**Android Sign**](https://docs.appcircle.io/workflows/android-specific-workflow-steps/android-sign)   | Required for signing the app; processes the app for signing. This step can be skipped if the app is already signed. |
```

| --------------------------------- | -------------------------------------------------------------------------------------------------------- |
| [**Xcodebuild for Devices**](https://docs.appcircle.io/workflows/ios-specific-workflow-steps#xcodebuild-for-devices-archive--export) | Builds the application in ARM architecture and generates an `IPA` file. |

<Screenshot url='https://cdn.appcircle.io/docs/assets/android-workflow-components-firebase-app-distribution_2.png'/>

#### For Android Flutter 

| Prerequisite Workflow Step        | Description                                                                                              |
| --------------------------------- | -------------------------------------------------------------------------------------------------------- |
| [**Flutter Build for Android**](https://docs.appcircle.io/workflows/flutter-specific-workflow-steps#flutter-build-for-android) | Generates the app required for the **Firebase App Distribution** step.                                                                           |

> | [**Android Sign**](https://docs.appcircle.io/workflows/android-specific-workflow-steps/android-sign)   | Required if using a signed app. Processes the output for signing. If already signed, this step can be skipped. |

Reviewer: ```suggestion
| [**Android Sign**](https://docs.appcircle.io/workflows/android-specific-workflow-steps/android-sign)   | Required for signing the app; processes the app for signing. This step can be skipped if the app is already signed. |
```

| `$GOOGLE_APPLICATION_CREDENTIALS` | Specifies the path of the Google Service Account JSON. Upload the service account as a file to your environment group and name it `GOOGLE_APPLICATION_CREDENTIALS`. **Select either a Firebase token or a Google Service account**. | Optional |
| `$AC_FIREBASE_RELEASE_NOTES`   | Specifies the release notes for this build. If you want to use a file for release notes, leave this field empty and configure the next section.                     | Optional |
| `$AC_FIREBASE_RELEASE_NOTES_PATH` | If you use the Publish Release Notes component before this step, `release-notes.txt` will be used as release notes.                                                  | Optional |
| `$AC_FIREBASE_GROUPS`          | Specifies the Firebase tester groups you want to invite.                                                                                               | Optional |
| `$AC_FIREBASE_EXTRA_PARAMETERS` | Specifies extra command line parameters. Enter `--debug` for debug mode.                                                                                              | Optional |

### Output Variables

> The **Firebase App Distribution** step does not produce any output variables. If the app is successfully distributed, the step will succeed. Otherwise, it will fail.

Reviewer: ```suggestion
The **Firebase App Distribution** step generates no output variables. The step succeeds if the app is distributed successfully; otherwise, it fails.
```
